src/gui/accountstate.cpp

```diff
@@ -47,47 +47,36 @@ Q_LOGGING_CATEGORY(lcAccountState, "gui.account.state", QtInfoMsg)
 
 // Returns the dialog when one is shown, so callers can attach to signals. If no dialog is shown
 // (because there is one already, or the new URL matches the current URL), a nullptr is returned.
-QDialog *AccountState::updateUrlDialog(const QUrl &newUrl)
+UpdateUrlDialog *AccountState::updateUrlDialog(const QUrl &newUrl)
 {
     // guard to prevent multiple dialogs
     if (_updateUrlDialog) {
         return nullptr;
     }
-    auto accept = [=] {
+
+    _updateUrlDialog = new UpdateUrlDialog(
+        tr("Url update requested for %1").arg(_account->displayName()),
+        tr("The url for %1 changed from %2 to %3, do you want to accept the changed url?").arg(_account->displayName(), _account->url().toString(), newUrl.toString()),
+        _account->url(),
+        newUrl,
+        ocApp()->activeWindow());
+
+    auto cleanup = [=]() {
```

Comment:
Better use:
`_updateUrlDialog->setAttribute(Qt::WA_DeleteOnClose);`

Reply:
That'd only work if the class inherited `QDialog` (or maybe even `QMessageBox`). The class's name is not ideal.

I'm not sure whether inheritance makes sense over composition here. We don't intend to extend the functionality of `QMessageBox`, but use one as-is with the existing interfaces. Therefore, composition makes more sense. Arguably, it requires some more boilerplate code...